`yay -Y --gendb` crashes with SIGSEGV

#### Affected Version
yay v5.608

#### Issue
`yay -Y --gendb` crashes with SIGSEGV

#### Steps to reproduce
```
$ yay -Y --gendb
==> Out Of Date AUR Packages: spotify
:: PKGBUILD up to date, Skipping (1/23): brave-bin
:: PKGBUILD up to date, Skipping (2/23): clipmenu
:: PKGBUILD up to date, Skipping (3/23): clipnotify
:: PKGBUILD up to date, Skipping (4/23): conan
:: PKGBUILD up to date, Skipping (5/23): dwm
:: PKGBUILD up to date, Skipping (6/23): ffmpeg-compat-54
:: PKGBUILD up to date, Skipping (7/23): google-chrome-beta
:: PKGBUILD up to date, Skipping (8/23): libavutil-52
:: PKGBUILD up to date, Skipping (9/23): libgcj17-bin
:: PKGBUILD up to date, Skipping (10/23): mons
:: PKGBUILD up to date, Skipping (11/23): openvpn-update-systemd-resolved
:: PKGBUILD up to date, Skipping (12/23): pdftk-bin
:: PKGBUILD up to date, Skipping (13/23): python-node-semver
:: PKGBUILD up to date, Skipping (14/23): python-patch
:: PKGBUILD up to date, Skipping (15/23): python-pluginbase
:: PKGBUILD up to date, Skipping (16/23): simple-mtpfs
:: PKGBUILD up to date, Skipping (17/23): spotify
:: PKGBUILD up to date, Skipping (18/23): st
:: PKGBUILD up to date, Skipping (19/23): tresorit
:: PKGBUILD up to date, Skipping (20/23): ttf-input
:: PKGBUILD up to date, Skipping (21/23): visual-studio-code-bin
:: Downloading PKGBUILD (22/23): wire-desktop-bin
:: PKGBUILD up to date, Skipping (23/23): yay
:: Parsing SRCINFO (1/23): brave-bin
:: Parsing SRCINFO (2/23): clipmenu
:: Parsing SRCINFO (3/23): clipnotify
:: Parsing SRCINFO (4/23): conan
:: Parsing SRCINFO (5/23): dwm
:: Parsing SRCINFO (6/23): ffmpeg-compat-54
:: Parsing SRCINFO (7/23): google-chrome-beta
:: Parsing SRCINFO (8/23): libavutil-52
:: Parsing SRCINFO (9/23): libgcj17-bin
:: Parsing SRCINFO (10/23): mons
:: Parsing SRCINFO (11/23): openvpn-update-systemd-resolved
:: Parsing SRCINFO (12/23): pdftk-bin
:: Parsing SRCINFO (13/23): python-node-semver
:: Parsing SRCINFO (14/23): python-patch
:: Parsing SRCINFO (15/23): python-pluginbase
:: Parsing SRCINFO (16/23): simple-mtpfs
:: Parsing SRCINFO (17/23): spotify
:: Parsing SRCINFO (18/23): st
:: Parsing SRCINFO (19/23): tresorit
:: Parsing SRCINFO (20/23): ttf-input
:: Parsing SRCINFO (21/23): visual-studio-code-bin
:: Parsing SRCINFO (22/23): wire-desktop-bin
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x1c0 pc=0x6636ba]

goroutine 1 [running]:
main.createDevelDB(0xc420090de0, 0xc420057d08)
        /home/xha/.cache/yay/yay/src/yay-5.608/vcs.go:53 +0x30a
main.handleYay(0xc4200a2430, 0x1)
        /home/xha/.cache/yay/yay/src/yay-5.608/cmd.go:290 +0xf0
main.handleCmd(0x0, 0x0)
        /home/xha/.cache/yay/yay/src/yay-5.608/cmd.go:155 +0x444
main.main()
        /home/xha/.cache/yay/yay/src/yay-5.608/main.go:204 +0x253
```

--- a/install.go
+++ b/install.go
@@ -72,7 +72,6 @@ func install(parser *arguments) error {
 		parser.targets.set(name)
 	}
 
-
 	requestTargets = parser.targets.toSlice()
 
 	if len(dt.Missing) > 0 {
@@ -491,6 +490,23 @@ func parseSRCINFOFiles(pkgs []*rpc.Pkg, srcinfos map[string]*gopkg.PKGBUILD, bas
 	}
 
 	return nil
+}
+
+func tryParsesrcinfosFile(pkgs []*rpc.Pkg, srcinfos map[string]*gopkg.PKGBUILD, bases map[string][]*rpc.Pkg) {
+	for k, pkg := range pkgs {
+		dir := config.BuildDir + pkg.PackageBase + "/"
+
+		str := bold(cyan("::") + " Parsing SRCINFO (%d/%d): %s\n")
+		fmt.Printf(str, k+1, len(pkgs), formatPkgbase(pkg, bases))
+
+		pkgbuild, err := gopkg.ParseSRCINFO(dir + ".SRCINFO")
+		if err != nil {
+			fmt.Printf("cannot parse %s skipping: %s\n", pkg.Name, err)
+			continue
+		}
+
+		srcinfos[pkg.PackageBase] = pkgbuild
+	}
 }
 
 func parseSRCINFOGenerate(pkgs []*rpc.Pkg, srcinfos map[string]*gopkg.PKGBUILD, bases map[string][]*rpc.Pkg) error {

--- a/vcs.go
+++ b/vcs.go
@@ -44,10 +44,13 @@ func createDevelDB() error {
 	bases := getBases(infoMap)
 
 	downloadPkgBuilds(info, sliceToStringSet(remoteNames), bases)
-	err = parseSRCINFOFiles(info, srcinfosStale, bases)
+	tryParsesrcinfosFile(info, srcinfosStale, bases)
 
 	for _, pkg := range info {
-		pkgbuild := srcinfosStale[pkg.PackageBase]
+		pkgbuild, ok := srcinfosStale[pkg.PackageBase]
+		if !ok {
+			continue
+		}
 
 		for _, pkg := range bases[pkg.PackageBase] {
 			updateVCSData(pkg.Name, pkgbuild.Source)

